pilot/pkg/config/clusterregistry/multicluster.go

```diff
@@ -107,10 +111,12 @@ func (m *Multicluster) AddMemberCluster(clientset kubernetes.Interface, clusterI
 // are removed.
 func (m *Multicluster) DeleteMemberCluster(clusterID string) error {
 
+	m.Lock()
 	m.serviceController.DeleteRegistry(clusterID)
 	close(m.rkc[clusterID].stopCh)
 	delete(m.rkc, clusterID)
```

Comment:
And here only need protect these two lines

Reply:
fixed